cpr/session.cpp

```diff
@@ -47,13 +48,15 @@ class Session::Impl {
     Url url_;
     Parameters parameters_;
     Proxies proxies_;
+    struct curl_httppost* formpost_;
```

Comment:
I don't disagree with your assessment here, just a minor point of clarification. `Session` already doesn't have copy semantics because it has an owning `std::unique_ptr` to `CurlHolder`. This can actually be leveraged to achieve memory safety for forms in an elegant way, and I'll explain it more in a comment below.